source/Halibut/DataStream.cs

```diff
@@ -133,11 +133,14 @@ public static DataStream FromStreamAsync(Stream source, Func<int, CancellationTo
         {
             updateProgress ??= (_, _) => Task.CompletedTask;
 
+#pragma warning disable CS0612
             return FromStream(source, 
                 i => updateProgress(i, CancellationToken.None).GetAwaiter().GetResult(),
                 updateProgress);
+#pragma warning restore CS0612
         }
 
+        [Obsolete]
```

Comment:
This might not need to be marked as obsolete since its only an issue if the `updateProgress` implementation is sync. But its probs safer to make everything async.